Bootstrap.php

```diff
@@ -939,6 +966,19 @@ public function onInitResourcePaypalRestClient()
     }
 
     /**
+     * Creates and returns the currency converter component.
+     *
+     * @return \Shopware_Components_Currency
+     */
+    public function onInitResourceCurrency()
+    {
+        require_once __DIR__ . '/Components/Currency.php';
+        $currency = new Shopware_Components_Paypal_Currency($this->Config());
```

Comment:
`Shopware_Components_Paypal_Currency` does not extend `Shopware_Components_Currency`.
This will lead to Exceptions in PHP >= 7 if users typhinted that class. 
Example:

```
public function myExample(Shopware_Components_Currency $currency) { ... }
```

You can simulate that behavior by setting `throwOnRecoverableError` to `true` in you `config.php`. See: throwOnRecoverableError
https://github.com/shopware/shopware/blob/5.1/engine/Shopware/Configs/Default.php#L51

```